packages/flutter/test/widgets/binding_live_test.dart

```diff
@@ -0,0 +1,61 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+// This file is for tests for WidgetsBinding that require a `LiveTestWidgetsFlutterBinding`.
+void main() {
+  LiveTestWidgetsFlutterBinding();
+  testWidgets('ReportTiming callback records the sendFramesToEngine when it was scheduled', (WidgetTester tester) async {
+    // This test needs LiveTestWidgetsFlutterBinding for multiple reasons.
+    //
+    // First, this was the environment that this bug was discovered.
+    //
+    // Second, unlike `AutomatedTestWidgetsFlutterBinding`, which overrides
+    // `scheduleWarmUpFrame` to execute the handlers synchronously,
+    // `LiveTestWidgetsFlutterBinding` still calls them asynchronously. This
+    // allows `runApp`, which also schedules a warm-up frame, to bind a widget
+    // without rendering a frame, which is needed to for `deferFirstFrame` to
+    // take effect.
+
+    // Before `testWidgets` executes the test body, it pumps a frame with a
+    // fixed dummy widget, then calls `resetFirstFrameSent`. The pumped frame
+    // schedules a reportTiming call that has yet to arrive.
+    //
+    // This puts the test in an inconsistent state: a reportTiming callback is
+    // supposed to happen only after a frame is rendered, but due to
+    // `resetFirstFrameSent`, the framework thinks no frames have been rendered.
+
+    expect(tester.binding.sendFramesToEngine, true);
+    // Push the widget with runApp instead of tester.pump, avoiding rendering a
+    // frame, which is needed for `deferFirstFrame` later to work.
+    runApp(const DummyWidget());
+    // Verify that no widget tree is built and nothing is rendered.
+    expect(find.text('First frame'), findsNothing);
+    // Defer the first frame, making sendFramesToEngine false, so that widget
+    // tree will be built but not sent to the engine.
+    tester.binding.deferFirstFrame();
+    expect(tester.binding.sendFramesToEngine, false);
+    // Wait for the reportTiming callback (which completes the future below) to
+    // run. If the reportTiming callback were to assume that
+    // `sendFramesToEngine` is true, the callback would crash.
+    await tester.binding.waitUntilFirstFrameRasterized;
+    expect(find.text('First frame'), findsOne);
```

Comment:
From the test failures: It looks like the test needs a `pump` call before this expect, which makes sense. Sorry to not include that in my previous suggestion.

Reply:
What do you think if I just revert to `    await tester.pump(const Duration(milliseconds: 1));`?

Reply:
I ended up including both.